src/unittest/test_random.cpp

```diff
@@ -62,18 +62,26 @@ void TestRandom::testPseudoRandom()
 {
 	PseudoRandom pr(814538);
 
-	for (u32 i = 0; i != 256; i++)
-		UASSERTEQ(s32, pr.next(), expected_pseudorandom_results[i]);
-
-	PseudoRandom pr2(0);
-	UASSERTEQ(int, pr2.next(), 0);
-	UASSERTEQ(int, pr2.next(), 21469);
-	UASSERTEQ(int, pr2.next(), 9989);
-
-	PseudoRandom pr3(-101);
-	UASSERTEQ(int, pr3.next(), 3267);
-	UASSERTEQ(int, pr3.next(), 2485);
-	UASSERTEQ(int, pr3.next(), 30057);
+	for (u32 i = 0; i != 128; i++)
+		UASSERTEQ(int, pr.next(), expected_pseudorandom_results[i]);
+
+	int	state = pr.getState();
+	PseudoRandom pr2(state);
+
+	for (u32 i = 128; i != 256; i++) {
+		UASSERTEQ(int, pr.next(), expected_pseudorandom_results[i]);
+		UASSERTEQ(int, pr2.next(), expected_pseudorandom_results[i]);
```

Comment:
for simplicitly I would leave the `expected_pseudorandom_results` test alone
and test the same situation as Lua here (`pt.next() == pr2.next()`)